subworkflows/local/taxid_reads.nf

```diff
@@ -0,0 +1,129 @@
+include { EXTRACT_VIRAL_TAXID as KRAKEN2_VIRAL_TAXID      } from '../../modules/local/extract_viral_taxid'
+include { EXTRACT_VIRAL_TAXID as CENTRIFUGE_VIRAL_TAXID   } from '../../modules/local/extract_viral_taxid'
+include { EXTRACT_VIRAL_TAXID as DIAMOND_VIRAL_TAXID      } from '../../modules/local/extract_viral_taxid'
+include { KRAKENTOOLS_EXTRACTKRAKENREADS                  } from '../../modules/nf-core/krakentools/extractkrakenreads/main'
+include { EXTRACTCENTRIFUGEREADS                          } from '../../modules/local/extractcentrifugereads'
+include { EXTRACTCDIAMONDREADS                            } from '../../modules/local/extractdiamondreads'
+
+workflow TAXID_READS {
+    params.taxid
+
+    take:
+    reads                   // channel:   [mandatory] [ meta, reads ]
+    kraken2_taxpasta        // channel:   [mandatory] [ meta, kraken2_taxpasta ]
+    kraken2_result          // channel:   [mandatory] [ meta, kraken2_result ]
+    kraken2_report          // channel:   [mandatory] [ meta, kraken2_report ]
+    centrifuge_taxpasta     // channel:   [mandatory] [ meta, centrifuge_taxpasta ]
+    centrifuge_result       // channel:   [mandatory] [ meta, centrifuge_result ]
+    centrifuge_report       // channel:   [mandatory] [ meta, centrifuge_report ]
+    diamond_taxpasta        // channel:   [mandatory] [ meta, diamond_taxpasta ]
+    diamond_tsv             // channel:   [mandatory] [ meta, diamond_tsv ]
+
+    main:
+    ch_versions = Channel.empty()
+
+    // extract kraken2 reads
+    if ( params.extract_kraken2_reads ) {
+        if ( params.taxid ) {
+            KRAKENTOOLS_EXTRACTKRAKENREADS(
+                params.taxid,
+                kraken2_result,
+                reads,
+                kraken2_report
+            )
+            ch_versions            = ch_versions.mix( KRAKENTOOLS_EXTRACTKRAKENREADS.out.versions.first() )
+
+        } else {
+            kraken2_taxids = KRAKEN2_VIRAL_TAXID( kraken2_taxpasta, kraken2_report )
+            combined_input = kraken2_taxids.viral_taxid
+                .map { meta,taxid -> [ meta.subMap( meta.keySet() - 'tool' ), taxid ] }
+                .splitText()
+                .combine( kraken2_result, by:0 )
+                .combine( reads, by:0 )
+                .combine( kraken2_report.map { meta, kraken2_report -> [ meta.subMap(meta.keySet() - 'tool'), kraken2_report ]}, by:0 )
+
+            ch_combined_input = combined_input.multiMap { meta,taxid,kraken2_result,reads,kraken2_report  ->
+                taxid: taxid.trim()
```

Comment:
Is `.trim()` used to parse each taxid in the list separately?

Reply:
trim the new line at the end of taxid.